tox.ini

```diff
@@ -1,14 +1,22 @@
 [tox]
-envlist = py38-django{32}-drf3
-    py39-django{32,41,42}-drf3
-    py310-django{32,41,42}-drf3
-    py311-django{41,42}-drf3
+envlist =
+    py38-django{32}
+    py39-django{32,41,42}
+    py310-django{32,41,42,50}
+    py311-django{41,42,50}
+    py312-django{42,50}
 
 [testenv]
-commands = python manage.py test
+set_env =
+    DJANGO_SETTINGS_MODULE=testsettings
+commands =
+    python \
+      -W error::DeprecationWarning \
+      -W error::PendingDeprecationWarning \
+      -m django test
```

Comment:
Using `set_env` + `python -m django` instead of having a `manage.py` script in the repo - fewer files are easier to manage.

Using `-W error` for deprecation warnings to ensure the package doesn’t trigger any warnings, so users don’t have to see them. I use this configuration on all of my open source projects.